src/librbd/migration/OpenSourceImageRequest.cc

```diff
@@ -116,9 +119,9 @@ void OpenSourceImageRequest<I>::finish(int r) {
   ldout(cct, 10) << "r=" << r << dendl;
 
   if (r < 0) {
-    delete *m_src_image_ctx;
```

Comment:
BTW (not related to this PR) I just observed a crash for a similar case in `api/Migration.cc:open_images`, after "open_images: failed to re-open destination image" error: the scope exit tried to close the unsuccessfully opened context.

The steps were: run `rbd migration abort` of a large image, and when it is in progress run `rbd migration status` in another window.

Reply:
Thanks -- I think I had fixed that in #38068